### PR TITLE
Add lock around protocol detection

### PR DESCRIPTION
**Pull request by narrieta**
Protocol detection creates a few files on disk, which need to be protected with a lock.

@byue - FYI

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1815)

### CODE REVIEW

File: tests/protocol/test_protocol_util.py [hunk 1]
```diff
@@ -44,29 +44,24 @@ def test_get_protocol_util_should_return_same_object_for_same_thread(self, _):
         self.assertEqual(protocol_util1, protocol_util2)
 
     def test_get_protocol_util_should_return_different_object_for_different_thread(self, _):
-        def get_util_obj(q, err):
+        protocol_util_instances = []
```
Comment (narrieta) [inline tests/protocol/test_protocol_util.py]: now the protocol object has a lock so it cannot be serialized into the queue

**Comment by byue** [comment]
Are there multiple readers? If so we could use read locks to allow multiple readers and write locks.

**Comment by narrieta** [comment]
@byue  This was an existing lock that we are putting back temporarily and will remove later again. Performance is not an issue here.

File: config/freebsd/waagent.conf [hunk 2]
```diff
@@ -48,10 +48,10 @@ ResourceDisk.Filesystem=ufs
 ResourceDisk.MountPoint=/mnt/resource
 
 # Create and use swapfile on resource disk.
-ResourceDisk.EnableSwap=n
+ResourceDisk.EnableSwap=y
```
Comment (larohra) [inline config/freebsd/waagent.conf]: I think changing the target branch picked up an unwanted commit with this PR

File: azurelinuxagent/common/protocol/util.py [hunk 3]
```diff
@@ -74,6 +74,7 @@ class ProtocolUtil(SingletonPerThread):
     """
 
     def __init__(self):
+        self._lock = threading.RLock()  # protects the files on disk created during protocol detection
```
Comment (pgombar) [inline azurelinuxagent/common/protocol/util.py]: What is the idea behind using a reentrant thread lock instead of a regular thread lock?

**Comment by narrieta** [comment]
Replaced by #1816, which will be part of release 2.2.47

**Pull request closed without merging**